### PR TITLE
fix(sql): fix incorrect unordered map creation check

### PR DESCRIPTION
**Pull request by puzpuzpuz**
`MapFactory#createUnorderedMap()` wasn't considering the size of the value correctly when deciding which map to create. This may have led to increased memory usage when running GROUP BY queries.

Other than that, fixes a few typos in comments and improves naming.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 46 / 48 (95.83%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/map/MapFactory.java|5|6|83.33%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|29|30|96.67%|
|:large_blue_circle:|io/questdb/std/Vect.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/map/OrderedMap.java|11|11|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlOptimiser.java [hunk 1]
```diff
@@ -139,6 +139,33 @@ public SqlOptimiser(
...
+            return false;
+        }
+        boolean appearsInArgs = false;
+        ObjList<QueryColumn> modelColumns = model.getColumns();
+        for (int i = 0, n = modelColumns.size(); i < n; i++) {
+            QueryColumn col = modelColumns.get(i);
+            switch (col.getAst().type) {
+                case FUNCTION:
+                case OPERATION:
+                    appearsInArgs |= searchExpressionNodeForAlias(col.getAst(), alias, sqlNodeStack);
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/SqlOptimiser.java:162]: nit: can be an early return here.

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 46 / 48 (95.83%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/map/MapFactory.java|5|6|83.33%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|29|30|96.67%|
|:large_blue_circle:|io/questdb/std/Vect.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/map/OrderedMap.java|11|11|100.00%|

**Comment by puzpuzpuz** [comment]
@ideoma thanks for the review!

**Pull request merged**